op-batcher/batcher/driver.go

```diff
@@ -272,16 +287,20 @@ func (l *BatchSubmitter) loop() {
 						l.Log.Error("Error closing the channel manager to handle a L2 reorg", "err", err)
 					}
 				}
-				l.publishStateToL1(queue, receiptsCh, true)
+				// on reorg we want to publish all pending state then wait until each result clears before resetting
+				// the state.
+				l.publishStateToL1(queue, receiptsCh)
+				if !l.Txmgr.IsClosed() {
+					queue.Wait()
+				} else {
```

Comment:
Looks like there's quite some duplication in in this reorg block and below in the shutdown `case`. Could this maybe be extracted into its own function?

Reply:
done